wallet_test.go

```diff
@@ -0,0 +1,164 @@
+		withdrawAmount Bitcoin
+		want           Bitcoin
+		wantErr        error
+	}{
+		{
+			name:           "Withdrawing positive value from positive balance",
+			w:              &Wallet{balance: 0.25},
+			withdrawAmount: 0.15,
+			want:           0.1,
+			wantErr:        nil,
+		},
+		{
+			name:           "Withdrawing from positive balance, but insufficient funds",
+			w:              &Wallet{balance: 0.5},
+			withdrawAmount: 0.75,
+			want:           0.5,
+			wantErr:        InsufficientFundsError,
+		},
+		{
+			name:           "Withdrawing negative value",
+			w:              &Wallet{balance: 0.01},
+			withdrawAmount: -0.0001,
+			want:           0.01,
+			wantErr:        NegativeInputError,
+		},
+		{
+			name:           "Withdrawing from negative balance",
+			w:              &Wallet{balance: -0.001},
+			withdrawAmount: 0.1,
+			want:           -0.001,
+			wantErr:        InsufficientFundsError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.w.Withdraw(tt.withdrawAmount); err != tt.wantErr || tt.w.balance != tt.want {
+				t.Errorf("Wallet.Withdraw() error = %v, wantErr = %v, want = %v, got = %v", err, tt.wantErr, tt.want, tt.w.balance)
+			}
+		})
+	}
+}
+
+func TestWallet_Deposit(t *testing.T) {
+	tests := []struct {
+		name          string
+		w             *Wallet
+		depositAmount Bitcoin
+		want          Bitcoin
+		wantErr       error
+	}{
+		{
+			name:          "Deposit positive value on positive balance",
+			w:             &Wallet{balance: 0.1},
+			depositAmount: 0.001,
+			want:          0.101,
+			wantErr:       nil,
+		},
+		{
+			name:          "Deposit positive value on negative balance",
+			w:             &Wallet{balance: -0.5},
+			depositAmount: 0.05,
+			want:          -0.45,
+			wantErr:       nil,
+		},
+		{
+			name:          "Deposit negative value",
+			w:             &Wallet{balance: 0.01},
+			depositAmount: -0.0001,
+			want:          0.01,
+			wantErr:       NegativeInputError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.w.Deposit(tt.depositAmount); err != tt.wantErr || tt.w.balance != tt.want {
+				t.Errorf("Wallet.Deposit() error = %v, wantErr = %v, want = %v, got = %v", err, tt.wantErr, tt.want, tt.w.balance)
+			}
+		})
+	}
+}
+
+func TestWallet_ConcurrentUse(t *testing.T) {
+	t.Run("ConcurrentUse", func(t *testing.T) {
+		w := Wallet{}
+		var wg sync.WaitGroup
+		numOfGourutinies := 200
+		wg.Add(numOfGourutinies)
+		for i := 0; i < numOfGourutinies; i++ {
+			go func(wg *sync.WaitGroup) {
+				w.Deposit(1)
+				defer wg.Done()
+			}(&wg)
+		}
+		wg.Wait()
+		numOfGourutinies = 90
+		wg.Add(numOfGourutinies)
+		for i := 0; i < numOfGourutinies; i++ {
+			go func(wg *sync.WaitGroup) {
+				w.Withdraw(2)
+				defer wg.Done()
+			}(&wg)
+		}
+		wg.Wait()
+		var wantBalance Bitcoin = 20
+		if w.balance != wantBalance {
+			t.Errorf(" want = %v, got = %v", wantBalance, w.balance)
+		}
+	})
```

Comment:
1. Your deposits run and finish, only then your withdrawals run and finish. They should run simultaneously. 
2. Add some newlines